api/tests/mocha/services/connected-user-log.spec.js

```diff
@@ -0,0 +1,83 @@
+const chai = require('chai');
+const sinon = require('sinon');
+const moment = require('moment');
+const connectedUserLogService = require('../../../src/services/connected-user-log');
+const db = require('../../../src/db');
+
+describe('Connected Users Log service', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('saveLog()', () => {
+    it('should save a log', () => {
+      const getStub = sinon.stub(db.medicLogs, 'get').returns(Promise.reject({ status: 404 }));
+      const putStub = sinon.stub(db.medicLogs, 'put').returns(Promise.resolve());
+      const expectedDoc = {
+        _id: 'connected-user-' + 'userXYZ',
+        user: 'userXYZ',
+        timestamp: 100
+      };
+
+      return connectedUserLogService
+        .save({user: 'userXYZ', timestamp: 100})
+        .then(() => {
+          chai.expect(getStub.called).to.be.true;
+          chai.expect(putStub.called).to.be.true;
+          chai.expect(putStub.args[0][0]).to.deep.include(expectedDoc);
```

Comment:
This is testing implementation details. Better would be to consider what your function interface and side effects are expected to be and only use those in your `expect` statements.

In this case the function is expected to have a certain return value (that's part of the interface) and to store a doc. Please add something to test the return value. For the expectation, there isn't a datastore here, so that stub will have to do and the `chai.expect(putStub.args[0][0]).to.deep.include(expectedDoc);` is fine here. I would drop the `chai.expect(getStub.called).to.be.true` and `chai.expect(putStub.called).to.be.true`.